### PR TITLE
Try to implement associations

### PR DESCRIPTION
**Pull request by Dagic-zewdu**
In this milestone of the school-library project
   - Create a class Classroom and give it a has-many relationship with students.
   - Create Book and Rental classes. Then add a many-to-many relationship between book and person using the Rental intermediate class.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  def person=(person)
+    @person = person
+    @person.rentals.push(self) unless @person.rentals.include?(self)
+  end
+
+  def book=(book)
+    @book = book
+    @book.rentals.push(self) unless @book.rentals.include?(self)
+  end
```
Comment (Lameck1) [inline rental.rb]: - [x] These methods are really not necessary. They are just repeating whatever is taking place in the initialize block. Please delete these two methods.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
+
+  def initialize(date, person, book)
+    @date = date
+
+    @person = person
+    @person.rentals.push(self) unless @person.rentals.include?(self)
```
Comment (Lameck1) [inline rental.rb]: - [x] The requirement states 👇🏼:
  
  > - Create the has-many side of Person and Rental (a person has many rentals).
  
  If a person should have many rentals, then this block `unless @person.rentals.include?(self)` negates this. Your implementation restricts a person to have only one rental. Please remove the unless block.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    @book = book
+    @book.rentals.push(self) unless @book.rentals.include?(self)
```
Comment (Lameck1) [inline rental.rb]: - [x] Same issues as above for book. Please remove the unless block.

File: person.rb [hunk 4]
```diff
@@ -1,23 +1,32 @@
...
+  def correct_name
+    @name
+  end
 
-  def of_age?
-    @age >= 18
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+    rental.person = self
   end
```
Comment (Lameck1) [inline person.rb]: - [x] In the above review comments, we realize that the Rental class is already taking care of pushing rentals to either a person or a book (`@person.rentals.push(self)` and `@book.rentals.push(self)` respectively). Therefore, whatever this method is doing currently is another repetition that will be a bug when the app is run.

  What we should be doing here is defining how a rental should be created by using the Rental class. Consider the example below:
  
  ```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
  ```

File: book.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+    rental.book = self
+  end
```
Comment (Lameck1) [inline book.rb]: - [x] Same as above, please refactor this accordingly. Get some hint on how the above example for the Person class has been implemented.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @Dagic-zewdu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, remember to tag me in your question so I can receive the notification.**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  attr_reader :person, :book
+
+  def initialize(date, person, book)
+    @date = date
+
+    @person = person
+    @person.rentals.push(self)
+
+    @book = book
+    @book.rentals.push(self)
```
Comment (Meltrust) [inline rental.rb:12]: - [ ] OPTIONAL: Helpful tip:  Easier way to call the #push method in Ruby:

```ruby
@person = person
person.rentals << self

@book = book
book.rentals << self

**Review by Meltrust — APPROVED** [review]
Hi @Dagic-zewdu,


## Wow, you did it 🎉
Brilliant
## Thank you for refactoring the rental and person methods, as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**